Orange/widgets/visualize/owmosaic.py

```diff
@@ -899,6 +899,13 @@ def draw_legend(x0_x1, y0_y1):
             self.Warning.no_valid_data.clear()
 
         if self.interior_coloring == self.PEARSON:
+            attrs = [attr for attr in attr_list if not data.domain[attr].values]
+            if attrs:
+                CanvasText(self.canvas,
+                           "Feature {} has no values".format(attrs[0]),
+                           (self.canvas_view.width() - 120) / 2,
+                           self.canvas_view.height() / 2)
+                return
```

Comment:
I would put this before `if`, that is, fail in both cases, with Pearson or with class distribution. The visualization does not make sense if any attribute has no values.